### PR TITLE
securitykey.js

### PR DESCRIPTION
**Pull request by XmerStudio**
Authenticate Users on the Systems/Applications you make available to them or make them unlock the data they store in their Storages/Servers with a key that only they can access. Requires Security Key (Hardware).

### CODE REVIEW

**Comment by veggiecan0419** [comment]
I have two suggestions. The bright green makes it a bit hard to read the white text, and the icon is hard to see in dark mode.

Maybe you could make the green a bit darker, and put the icon into a green circle?

File: extensions/XmerOriginals/securitykey.js [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+        color1: "#5ef726",
+        color2: "#1fbd00",
+        color3: "#0ecc00",
+        id: "securitykey",
+        name: "Security Key",
+        blocks: [
+          {
+            opcode: "isSupported",
+            blockType: Scratch.BlockType.BOOLEAN,
+            text: "Is security key supported?",
```
Comment (veggiecan0419) [inline extensions/XmerOriginals/securitykey.js:31]:
```suggestion
            text: "security key supported?",
```

File: extensions/XmerOriginals/securitykey.js [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+          },
+          {
+            opcode: "requestAuthentication",
+            blockType: Scratch.BlockType.COMMAND,
+            text: "Request security key authentication",
```
Comment (veggiecan0419) [inline extensions/XmerOriginals/securitykey.js:36]:
```suggestion
            text: "request security key authentication",
```

File: extensions/XmerOriginals/securitykey.js [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+          },
+          {
+            opcode: "isAuthenticated",
+            blockType: Scratch.BlockType.BOOLEAN,
+            text: "Is authenticated?",
```
Comment (veggiecan0419) [inline extensions/XmerOriginals/securitykey.js:41]:
```suggestion
            text: "is authenticated?",
```

File: extensions/XmerOriginals/securitykey.js [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+          },
+          {
+            opcode: "isAuthenticationFailed",
+            blockType: Scratch.BlockType.BOOLEAN,
+            text: "Is authentication failed?",
```
Comment (veggiecan0419) [inline extensions/XmerOriginals/securitykey.js:46]:
```suggestion
            text: "authentication failed?",
```

File: extensions/XmerOriginals/securitykey.js [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+          },
+          {
+            opcode: "getStatus",
+            blockType: Scratch.BlockType.REPORTER,
+            text: "Status",
```
Comment (veggiecan0419) [inline extensions/XmerOriginals/securitykey.js:51]:
```suggestion
            text: "status",
```

**Review by veggiecan0419 — CHANGES_REQUESTED** [review]
This would make it match the other extensions

File: extensions/XmerOriginals/securitykey.js (hunk 3 above)
Comment (veggiecan0419) [reply extensions/XmerOriginals/securitykey.js:41]: Or maybe:
```suggestion
            text: "authenticated?",
```

**Review by veggiecan0419 — APPROVED** [review]

**Comment by veggiecan0419** [comment]
@XmerStudio 

Something happened to the icon...

Also, you should probably use a separate block icon from menu icon. The icon was only a problem in the menu.

File: extensions/XmerOriginals/securitykey.js
Comment (veggiecan0419) [inline extensions/XmerOriginals/securitykey.js:1]: I guess...

**Comment by veggiecan0419** [comment]
Sorry, last thing, I forgot, the id should have your username (eg. `XmerSecurityKey`)

**Comment by GarboMuffin** [comment]
this is client-side authentication, has the same problem as sharkpool captcha extension in not providing any actual security

**Pull request closed without merging**